### PR TITLE
widget color changed

### PR DESCRIPTION
**Pull request by matallo**
re: https://github.com/CartoDB/cartodb/issues/12394

when reloading a map, and selecting a category in the category widget it won't color the category blue but leave it in the default green

this is caused because the widget color is persisted, and currently there's no way to know if the widget color has been changed

not really sure about the new `widget_color_changed` attribute but I couldn't think of a better solution, that is backwards compatible too

this needs changes in deep insights: https://github.com/CartoDB/deep-insights.js/pull/556

**acceptance**

- default category widgets should change the color to blue when selecting the category
- autostyle widgets should leave the autostyle category color when selecting the category
- widgets with color changed should leave the widget color when selecting the category

- newly added widgets should work properly
- reload and widgets should work properly
- check thouroughly previously created widgets still work after deploying the new changes

/cc discussed with @noguerol

### CODE REVIEW

File: lib/assets/core/test/spec/cartodb3/deep-insights-integrations.spec.js [hunk 1]
```diff
@@ -563,6 +563,21 @@ describe('deep-insights-integrations/dii', function () {
...
         });
       });
+
+      describe('color', function () {
+        it('should set widget color changed to true', function () {
+          expect(this.model.get('widget_color_changed')).toBe(false);
+          this.model.set({
+            widget_style_definition: {
+              color: {
+                fixed: '#fabada',
```
Comment (rjimenezda) [inline lib/assets/core/test/spec/cartodb3/deep-insights-integrations.spec.js]: Best color 👌

**Comment by nobuti** [comment]
I found something, not sure if we should tackle it in this PR. The fill colour is gone once you check the custom colours, even if autostyle is off:

widget-color

**Comment by matallo** [comment]
can't reproduce

widgetscolors

I'm going to deploy again in ded03

**Comment by nobuti** [comment]
Reproduced in ded03, using this https://matataded03team.carto-staging.com/u/matataded03owner/builder/5f42b26e-6210-11e7-90a9-040146752f01 visualisation.

**Comment by matallo** [comment]
redeployed in staging and couldn't manage to reproduce it again, merging!

**Pull request merged**